### PR TITLE
more understandable errors from merkledag decoding

### PR DESCRIPTION
**Pull request by whyrusleeping**
This addresses bug #831

### CODE REVIEW

File: merkledag/coding.go [hunk 1]
```diff
@@ -92,7 +93,10 @@ func (n *Node) Encoded(force bool) ([]byte, error) {
 
 // Decoded decodes raw data and returns a new Node instance.
 func Decoded(encoded []byte) (*Node, error) {
-	n := &Node{}
+	n := new(Node)
 	err := n.Unmarshal(encoded)
-	return n, err
+	if err != nil {
+		return nil, errors.New("incorrectly formatted merkledag node")
```
Comment (cryptix) [inline merkledag/coding.go]: might want to wrap the original error as well?

Comment (whyrusleeping) [inline merkledag/coding.go]: I thought about that, but this is an error that will be passed to the user directly (in most cases). Whatever error Unmarshal spits out is not going to be something we want the users to see

Comment (jbenet) [inline merkledag/coding.go]: i think cryptix has a point-- how will they fix their formatting error without feedback? maybe: 

```go
fmt.Errorf("incorrectly formatted merkledag node: %s", err)
```

Comment (whyrusleeping) [inline merkledag/coding.go]: hrm... okay.

**Comment by whyrusleeping** [comment]
@jbenet hows that look?

**Pull request merged**